### PR TITLE
Add a button to copy the action-plan to clipboard.

### PR DESCRIPTION
**Pull request by neutrino**
**Description:**
Allow the user to copy the action plan with a click of a button.

Closes : #105 

Screen Shot 2021-12-14 at 18 20 29

___

I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/pull_request_template.md).

### CODE REVIEW

**Review by mateusdeap — APPROVED** [review]
All looks good to me

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css" rel="stylesheet">
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.8/clipboard.min.js"></script>
```
Comment (arielj) [inline app/views/layouts/application.html.erb:11]: maybe we can include this file in the application.js? so it doesn't have to be downloaded in parallel

**Review by arielj — COMMENTED** [review]
hey! just added some refactor ideas, but the copy to clipboard works as expected :+1: 

One suggestion, it's easier to test your branch locally if you branch on the main fastruby/points repo, so we don't have to clone your repo to test things, you should have access to do that

File: app/assets/javascripts/action_plan.js [hunk 2]
```diff
@@ -7,4 +7,6 @@ document.addEventListener("turbolinks:load", function() {
       el.innerText = `${prefix}${suffix} `
     })
   })
+
+  new ClipboardJS('.btn-clipboard');
```
[the remark this answers is not in the document]
Comment (neutrino) [reply app/assets/javascripts/action_plan.js:11]: Sure, I will look into it.

**Review by arielj — COMMENTED** [review]
hey @neutrino one more change here. It would be good to have a test for this. No need to test the actual copy to clipboard functionality, but I think it's a good idea to test that the button is there and the setup of the button is correct (like... making sure the button has the btn-clipboard class and that there's a div with the id defined in the `data-clipboard-target`), in case somebody wants to refactor this view for any reason and breaks the functionality.

The rest is just relaying on ClipboardJS so I wouldn't care about testing what happens when clicking the button.

**Comment by neutrino** [comment]
@arielj : Sure, I will add that.

**Review by arielj — APPROVED** [review]
LGTM

**Pull request merged**